force-app/main/default/classes/ClassesObjectsMethods.cls

```diff
@@ -126,16 +157,26 @@ public with sharing class ClassesObjectsMethods {
         List<String> scientistNames = new List<String>{'Tim Berners-Lee', 'Alan Turing', 'Grace Hopper', 'Donald Knuth', 'Guido van Rossum', 'Ken Thompson', 'Stephen Hawking'};
 
         // The variable to store the concatenated string
+        String namesContainingG = '';
 
         // Loop through the list of scientists' names
-
-            // If the name doesn't contain the letter 'G', skip this iteration
-
-            // Add the name to the result string, followed by a comma
+        for(String name : scientistNames){
+            if(!name.toLowerCase().contains('g')) {    
+             // If the name doesn't contain the letter 'G', skip this iteration; 
+             //Names of the scientists are converted to lower case names to not omitt names trough different spelling
+            continue;
+            }
+         
+        // Add the name to the result string, followed by a comma
+        namesContainingG += name + ',';
```

Comment:
No need to use a continue statement here. If the name is a g name add it to the list. Otherwise you are going to continue to the next iteration regardless. 

```
 public static String findTheScientist() {
        // The list of scientists' names
        List<String> scientistNames = new List<String>{
                'Tim Berners-Lee', 'Alan Turing', 'Grace Hopper', 'Donald Knuth', 'Guido van Rossum', 'Ken Thompson', 'Stephen Hawking'
        };
        String result = '';
        for (String name : scientistNames) {
            if (name.toLowerCase().contains('g')) {
                result += name + ',';
            }
        }
        return result.removeEnd(',');
    }
    ```